### ISSUE TITLE
Be nicer when reporting version conflicts

### ISSUE THREAD

**Issue by gforcada** [buildout/buildout#276]
It would  be really really helpful that whenever a version conflict the package that has the version conflict is reported.

Like on https://github.com/collective/experimental.safe_html_transform/pull/43#issuecomment-149902204 where Prakhar was trying to fix the version for plone.app.z3cform although the problem is on another package that has this dependency specified on its setup.py.

**Comment by reinout** [comment]
I've got the same kind of problems myself (mostly because colleagues get similar error messages and we're reduced to grepping for a version number in a pile of eggs, literally, to find out which package it is).

So... I've spend *quite* some time improving and trying to improve the logging buldout provides. Biggest problem is that some of the dependency handling is done inside setuptools and some of the errors come from there. (At least that's what I remember now).

The only solution I saw at the time was to let buildout maintain its own list of requirements and dependencies. So: query all the packages that are being installed for their dependency requirements. And once something goes wrong, to look up the package in our own lists and print out the information.

I *do* fear we'll still have the problem that we're thinking package A is being installed while setuptools is busy with B, a dependency of A, and reports the version conflict.
